week2/assignment2/index.js

```diff
@@ -0,0 +1,187 @@
+  
+    const checkboxes = todoBody.querySelectorAll('input[type="checkbox"]');
+    const allChecked = checkboxes.length && [...checkboxes].every(cb => cb.checked);
+    selectAll.checked = allChecked;
+  
+    enableDragAndDrop();
+}
+  
+function enableDragAndDrop() {
+    let dragged;
+  
+    todoBody.querySelectorAll('tr').forEach(row => {
+      row.addEventListener('dragstart', () => {
+        dragged = row;
+        row.classList.add('dragging');
+      });
+  
+      row.addEventListener('dragend', () => {
+        dragged.classList.remove('dragging');
+        dragged = null;
+    
+        const ids = [...todoBody.querySelectorAll('tr')].map(tr => +tr.dataset.id);
+        todos.sort((a, b) => ids.indexOf(a.id) - ids.indexOf(b.id));
+        saveTodos();
+      });
+  
+      row.addEventListener('dragover', e => {
+        e.preventDefault();
+        const after = [...todoBody.children].find(child => {
+          return e.clientY <= child.getBoundingClientRect().top + child.offsetHeight / 2;
+        });
+        if (after) todoBody.insertBefore(dragged, after);
+        else todoBody.appendChild(dragged);
+      });
+    });
+}
+  
+addBtn.addEventListener('click', () => {
+    const title = todoInput.value.trim();
+    const priority = parseInt(prioritySelect.value);
+  
+    if (!title || isNaN(priority)) {
+      alert('할 일과 중요도를 모두 입력해주세요!');
+      return;
+    }
+  
+    todos.push({ id: Date.now(), title, completed: false, priority });
+    saveTodos();
+    render();
+  
+    todoInput.value = '';
+    todoInput.placeholder = '할 일 입력';
+    prioritySelect.value = '';
+});
+  
+selectAll.addEventListener('change', () => {
+    const checked = selectAll.checked;
+    todos.forEach(t => t.completed = checked);
+    saveTodos();
+    render();
+});
+  
+filterButtons.forEach(btn => {
+    btn.addEventListener('click', () => {
+      currentFilter = btn.dataset.filter;
+      render();
+    });
+});
+  
+priorityDropdown.querySelector('button').addEventListener('click', () => {
+    priorityDropdown.classList.toggle('active');
+});
+  
+priorityOptions.forEach(opt => {
+    opt.addEventListener('click', () => {
+      currentPriority = parseInt(opt.dataset.priority);
+      render();
+      priorityDropdown.classList.remove('active');
+    });
+});
+
+function getCheckedTodoIds() {
+    return [...todoBody.querySelectorAll('tr')].filter(row =>
+      row.querySelector('input[type="checkbox"]').checked
+    ).map(row => parseInt(row.dataset.id));
+}
+  
+document.getElementById('delete-btn').addEventListener('click', () => {
+    const idsToDelete = getCheckedTodoIds();
+    if (idsToDelete.length === 0) return alert("삭제할 항목을 선택하세요!");
+  
+    todos = todos.filter(todo => !idsToDelete.includes(todo.id));
+    saveTodos();
+    render();
+});
+  
+document.getElementById('mark-done-btn').addEventListener('click', () => {
+    const idsToMark = getCheckedTodoIds();
+    if (idsToMark.length === 0) return alert("완료할 항목을 선택하세요!");
```

Comment:
```suggestion
    if{
    (idsToMark.length === 0) return alert("완료할 항목을 선택하세요!");
    } 
```

개인적으로 이런 방법이 더 가독성이 좋다고 생각해요